### PR TITLE
Implement search with typeahead

### PR DESCRIPTION
**Pull request by guybrush**

### CODE REVIEW

File: cmd/explorer/main.go [hunk 1]
```diff
@@ -96,6 +96,7 @@ func main() {
 		router.HandleFunc("/validators/data/active", handlers.ValidatorsDataActive).Methods("GET")
 		router.HandleFunc("/validators/data/ejected", handlers.ValidatorsDataEjected).Methods("GET")
 		router.HandleFunc("/search", handlers.Search).Methods("POST")
+		router.HandleFunc("/searchAhead/{type}/{search}", handlers.SearchAhead).Methods("GET")
```
Comment (ppratscher) [inline cmd/explorer/main.go]: I don't believe we need a dedicated enpoint for the new search. You should be able to use the old /search endpoint with the new parameters.

File: handlers/search.go [hunk 2]
```diff
@@ -13,17 +17,67 @@ func Search(w http.ResponseWriter, r *http.Request) {
...
-		http.Redirect(w, r, "/validator/" + search, 301)
+		http.Redirect(w, r, "/validator/"+search, 301)
 	} else {
 		http.Error(w, "Not found", 404)
 	}
-}
\ No newline at end of file
+}
+
+type SearchAheadBlocksResult []struct {
```
Comment (ppratscher) [inline handlers/search.go]: Please move type definitions into the dedicated types package

File: cmd/explorer/main.go (hunk 1 above)
Comment (guybrush) [reply cmd/explorer/main.go]: I am now doing:
```golang
router.HandleFunc("/search", handlers.Search).Methods("POST")
router.HandleFunc("/search/{type}/{search}", handlers.SearchAhead).Methods("GET")
```

File: handlers/search.go (hunk 2 above)
Comment (guybrush) [reply handlers/search.go]: Moved it to `types/templates.go`

**Pull request merged**